### DISCUSSION TITLE
WaveShare WS170120 Touchscreen

### DISCUSSION THREAD

**Discussion by Axialis** [rsta2/circle#478] [Development]
Hi. Can I get a way to add WaveShare WS170120 touchscreen support to code? I'm using the sample to calibrate the display, but I got the error “usbdev1-4: Cannot get configuration descriptor (short)” and kernel: Touchscreen not found. But Pi OS works with it correctly. I tried use command touchscreen=3000 in cmdline.txt, but not result. Where do I start looking for the problem, I don't know where to begin.

> [    5.670615] usb 1-1.4: New USB device found, idVendor=0eef, idProduct=0005, bcdDevice= 1.11
> [    5.670646] usb 1-1.4: New USB device strings: Mfr=1, Product=2, SerialNumber=3
> [    5.670660] usb 1-1.4: Product: WS170120
> [    5.670672] usb 1-1.4: Manufacturer: WaveShare
> [    5.670683] usb 1-1.4: SerialNumber: 220211
> [    5.693731] input: WaveShare WS170120 Touchscreen as /devices/platform/soc/3f980000.usb/usb1/1-1/1-1.4/1-1.4:1.0/0003:0EEF:0005.0003/input/input8
> [    5.694689] hid-generic 0003:0EEF:0005.0003: input,hiddev97,hidraw2: USB HID v1.11 Device [WaveShare WS170120] on usb-3f980000.usb-1.4/input0

![IMG_3214](https://github.com/user-attachments/assets/8669716e-9ad8-407e-822d-6cbb2819c276)

**Comment by rsta2** [comment]
What happens, when you connect the touchscreen directly to the root port (without hub)?

**Reply by Axialis** [reply]
I'll try the options. Because right now I'm using https://www.waveshare.com/wiki/Zero-DISP-7A

**Reply by Axialis** [reply]
Yes, it works now, I removed the FE1.1s chip and used short wires instead of the chip.

**Reply by rsta2** [reply]
I did not know, that this is an compound device with integrated USB hub. I tried to reproduce the problem with [this device](https://www.waveshare.com/wiki/5inch_HDMI_LCD_%28B%29) and an external USB hub. This touchscreen seems to have the same display controller (same vendor/device ID). I can reproduce the problem, when the hub is bus powered, but not when it is self powered (external power supply). The power supply seems to be critical.

**Reply by Axialis** [reply]
I will keep this point in mind. Thank you for your help and efforts in resolving this issue! It has moved me forward from my dead end 👍🏻

**Reply by jsanchezv** [reply]
> I did not know, that this is an compound device with integrated USB hub. I tried to reproduce the problem with [this device](https://www.waveshare.com/wiki/5inch_HDMI_LCD_%28B%29) and an external USB hub. This touchscreen seems to have the same display controller (same vendor/device ID). I can reproduce the problem, when the hub is bus powered, but not when it is self powered (external power supply). The power supply seems to be critical.

Try again joining the Grounds when the hub is self powered. I've seen this problem with a self powered flow meter connected to a USB to 232 serial converter. Joining Grounds the problem go away.

**Reply by rsta2** [reply]
> Try again joining the Grounds when the hub is self powered. I've seen this problem with a self powered flow meter connected to a USB to 232 serial converter. Joining Grounds the problem go away.

This makes sense. Thanks for info!
